packages/shared/lib/components/ToggleTokens/index.type.ts

```diff
@@ -0,0 +1,17 @@
+import { SpaceProps } from 'antd';
+
+export type ToggleTokensOptionsType = Array<
+  | {
+      label: React.ReactNode;
+      value: string | number;
+    }
+  | string
+>;
+
+export type ToggleTokensProps<V = string> = {
+  options: ToggleTokensOptionsType;
+  value?: V;
+  onChange?: (val?: V) => void;
+  multiple?: boolean;
+  withCheckbox?: boolean;
+} & Omit<SpaceProps, 'onChange'>;
```

Comment:
这个倒是可以用联合类型优化一下。但是用联合类型可读性下降了了。你看一下要不要优化吧。
```suggestion
type ToggleTokensProps<V = string> = {
  withCheckbox?: boolean;
}

type A<V = string> = {
  value?: V[];
  onChange?: (val?: V[]) => void;
  multiple: true;
}

type B<V = string> = {
  value?: V;
  onChange?: (val?: V) => void;
  multiple?: false;
}

type C<V = string> = ToggleTokensProps<V> & A<V> | ToggleTokensProps<V> & B<V>
```

Reply:
这个联合类型好像依旧无法解决使用时类型的推断问题。
![image](https://github.com/actiontech/dms-ui/assets/42765421/21c807cc-9dcd-4f06-a428-29569bec0e42)

Reply:
。。应该是可以的，上面是 value 类型固定为 string 的原因。。